source/blog/2015-06-17-ember-data-1-13-released.md

```diff
@@ -0,0 +1,376 @@
+If you have a custom `normalize()` method you need to update this to return a full JSON-API Document ([http://jsonapi.org/format/#document-top-level]()) with the primary resource object ([http://jsonapi.org/format/#document-resource-objects]()) located in `data`.
+
+##### Opt into the new Serializer API
+To tell Ember Data that you are ready to use the new Serializer API, add `isNewSerializerAPI: true` when extending `JSONSerializer` or `RESTSerializer`. The new `JSONAPISerializer` uses the new Serializer API by default. The `isNewSerializerAPI` flag is **only** required for Ember Data 1.13 and will be removed in Ember Data 2.0.
+
+### New Errors API
+Error handling is an important part of any ambitious application. Ember-Data had a support for what we called invalid errors from the very beginning. Invalid errors are a type of errors that can be corrected by the user, means a "correct and retry" cycle is encouraged for this type of errors. But there a other type of errors that can occur in the adapter. A network can be down or you can reach the quota on your `localStorage`. We wanted to provide some better semantics to adapter authors on how to handle this type of errors and expos an api to application authors to react on them.
+
+In Ember-Data 1.13 we introduced a new adapter hook: `handleResponse`. At the same
+time we deprecating `ajaxSuccess` and `ajaxError`.
+
+There been quite a bit of discussion on what an error in the context of an adapter actually mean and how much control should we allow to the application on deciding given a particular response. At some point we made changes to support returning `InvalidError` from `ajaxSuccess`. This is why today we replacing both hooks (`ajaxSuccess` and `ajaxError`) with just one (`handleResponse`). The default implementation will use a couple of new hooks (`isSuccess` and `isInvalid`) to decide if a response is a success or an error of certain type. It allows for more control over how your application handles errors.
+
+We also wanted for quite some time to abstract `jQuery.ajax` semantics. We used to reject from the adapter with `jqXHR` object and leave the application to deal with it. This strategy have a few drawbacks. It made life hard for application authors by making switch from one adapter to another a lot of work. It also made life of adapter authors hard by failing to provide them a common set of error types that can be shared across different backends. A `TimeoutError` for example can express a common type of error across a wide range of backends.
+
+We made a decision in Ember-Data to use JSON-API format as internal representation of the data. Naturally, we decided to do the same for errors. JSON-API have specified an error object format. Starting with Ember-Data 1.13 we are using JSON-API format to communicate errors from the adapter to the store. Also RESTAdapter and JSONAPISerializer supports out of the box JSON-API errors objects in the error response payload.
+
```

Comment:
I added some examples of old / new payload:

 It means that the Ruby on Rails inspired format we were using as payload for `InvalidError` is replaced with proper JSON-API objects. Old format is still supported for now with a deprecation warning.

Deprecated format:
```js
new DS.InvalidError({
  first_name: [‘is invalid’]
});
```
New format:
```js
new DS.InvalidError([
  {
    source: { pointer: ‘data/attributes/first_name’ },
    details: ‘is invalid’
  }
]);